compaction/compaction_manager.cc

```diff
@@ -1470,10 +1470,18 @@ class compaction_manager::cleanup_sstables_compaction_task : public compaction_m
 bool needs_cleanup(const sstables::shared_sstable& sst,
                    const dht::token_range_vector& sorted_owned_ranges,
                    schema_ptr s) {
-    auto first = sst->get_first_partition_key();
-    auto last = sst->get_last_partition_key();
-    auto first_token = dht::get_token(*s, first);
-    auto last_token = dht::get_token(*s, last);
+    const auto& first = sst->get_first_decorated_key();
+    const auto& last = sst->get_last_decorated_key();
+    dht::token first_token;
+    dht::token last_token;
+    // avoid partitioner string compare if schema is the same
+    if (sst->get_schema() == s || sst->get_schema()->get_partitioner() == s->get_partitioner()) [[likely]] {
```

Comment:
I don't think we even support changing the partitioner currently. It is hardcoded into the schema.

Reply:
so maybe we should just a debug level assert to cement this assumption

Reply:
We rely on this fact throughout the codebase. I just don't think we should start to add asserts for it now. It is like adding an assert that the primary key type didn't change. It is pointless IMO.
Also, these keys are decorated after the sstables are loaded, they are not stored in a decorated form on disk.

Reply:
okay, I'll simplify this patch then.

Reply:
Done in v4.